src/Campaign.ts

```diff
@@ -0,0 +1,165 @@
+import i18next from 'i18next'
+import { CorruptionLoadout, type Corruptions } from './Corruptions'
+import { player } from './Synergism'
+import { DOMCacheGetOrSet } from './Cache/DOM'
+import { IconSets } from './Themes'
+
+export type AscensionModifiers = 'GlobalSpeed'
+
+export type CampaignLoadout = Partial<Corruptions>
+export type CampaignModifiers = Partial<Record<AscensionModifiers, number>>
+
+export type CampaignKeys = 'test1' | 'test2' | 'test3'
+
+export interface ICampaignManagerData {
+  currentCampaign?: CampaignKeys | undefined
+  campaigns?: Record<CampaignKeys, number>
+}
+
+export interface ICampaignData {
+  campaignCorruptions: Partial<Corruptions>
+  campaignModifiers: CampaignModifiers
+  limit: number
+  isMeta: boolean
+}
+
+export class CampaignManager {
+  private totalCampaignTokens: number
+  private currentCampaign: Campaign | undefined
+  private campaigns!: Record<CampaignKeys, Campaign>
+
+  constructor (campaignManagerData?: ICampaignManagerData) {
+    this.campaigns = {} as Record<CampaignKeys, Campaign>
```

Comment:
casting shouldn't be needed. Both branches use mostly the same logic. Also you can use Object.entries()